code/game/objects/items/tools/weldingtool.dm

```diff
@@ -114,20 +114,30 @@
 	dyn_explosion(T, plasmaAmount/5)//20 plasma in a standard welder has a 4 power explosion. no breaches, but enough to kill/dismember holder
 	qdel(src)
 
-/obj/item/weldingtool/attack(mob/living/carbon/human/H, mob/user)
-	if(!istype(H))
+/obj/item/weldingtool/use_tool(atom/target, mob/living/user, delay, amount, volume, datum/callback/extra_checks)
+	target.add_overlay(GLOB.welding_sparks)
+	. = ..()
+	target.cut_overlay(GLOB.welding_sparks)
+
+/obj/item/weldingtool/attack(mob/living/carbon/human/target, mob/user)
+	if(!istype(target))
 		return ..()
 
-	var/obj/item/bodypart/affecting = H.get_bodypart(check_zone(user.zone_selected))
+	var/obj/item/bodypart/attackedLimb = target.get_bodypart(check_zone(user.zone_selected))
 
-	if(affecting && (!IS_ORGANIC_LIMB(affecting)) && user.a_intent != INTENT_HARM)
-		if(src.use_tool(H, user, 0, volume=50, amount=1))
-			if(user == H)
-				user.visible_message("<span class='notice'>[user] starts to fix some of the dents on [H]'s [parse_zone(affecting.body_zone)].</span>",
-					"<span class='notice'>You start fixing some of the dents on [H == user ? "your" : "[H]'s"] [parse_zone(affecting.body_zone)].</span>")
-				if(!do_mob(user, H, 50))
-					return
-			item_heal_robotic(H, user, 15, 0)
+	if(attackedLimb && (!IS_ORGANIC_LIMB(attackedLimb)) && user.a_intent != INTENT_HARM)
+		if(!tool_start_check(user, amount=1))
+			return
+		var/selfFix=(target == user)
+		user.visible_message("<span class='notice'>[user] starts to fix some of the dents on [target]'s [parse_zone(attackedLimb.body_zone)].</span>",
+			"<span class='notice'>You start fixing some of the dents on [selfFix ? "your" : "[target]'s"] [parse_zone(attackedLimb.body_zone)].</span>")
+		if(selfFix)
+			if(!use_tool(target, user, delay=50, amount=0, volume=30)) //Setting this to 0 because afterattack seems to be handling fuel consumption already. No need to take the fuel twice.
+				return
+		else
+			if(!use_tool(target, user, delay=5, amount=0, volume=25))
+				return
```

Comment:
just use && here, no need to make another "step" if contains nothing
also you can just use a tertiary statement here like i told you